### ISSUE TITLE
Only create copy_nop instructions for types for which an encoding exists

### ISSUE THREAD

**Issue by bjorn3** [CraneStation/cranelift#779]
```
thread 'main' panicked at 'assertion failed: ok', cranelift-codegen/src/regalloc/reload.rs:236:21
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:39
   1: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/sys_common/backtrace.rs:59
             at src/libstd/panicking.rs:197
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:211
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:474
   5: std::panicking::begin_panic
             at /rustc/3c235d5600393dfe6c36eeed34042efad8d4f26e/src/libstd/panicking.rs:408
   6: cranelift_codegen::regalloc::reload::Context::visit_inst
             at cranelift-codegen/src/regalloc/reload.rs:236
   7: cranelift_codegen::regalloc::reload::Context::visit_ebb
             at cranelift-codegen/src/regalloc/reload.rs:133
   8: cranelift_codegen::regalloc::reload::Context::run
             at cranelift-codegen/src/regalloc/reload.rs:118
   9: cranelift_codegen::regalloc::reload::Reload::run
             at cranelift-codegen/src/regalloc/reload.rs:86
  10: cranelift_codegen::regalloc::context::Context::run
             at cranelift-codegen/src/regalloc/context.rs:164
  11: cranelift_codegen::context::Context::regalloc
             at cranelift-codegen/src/context.rs:308
  12: cranelift_codegen::context::Context::compile
             at cranelift-codegen/src/context.rs:145
  13: cranelift_codegen::context::Context::compile_and_emit
             at cranelift-codegen/src/context.rs:104
```

Caused by #773

<details>

```
target x86_64

function u0:0(i64, i64, i64) system_v {
    ss0 = explicit_slot 16
    ss1 = explicit_slot 1
    ss2 = explicit_slot 16
    ss3 = explicit_slot 1
    ss4 = explicit_slot 16
    ss5 = explicit_slot 8
    ss6 = explicit_slot 16
    ss7 = explicit_slot 16
    ss8 = explicit_slot 16
    ss9 = explicit_slot 16
    ss10 = explicit_slot 16
    ss11 = explicit_slot 16
    ss12 = explicit_slot 16
    ss13 = explicit_slot 16
    ss14 = explicit_slot 16
    ss15 = explicit_slot 16
    ss16 = explicit_slot 16
    ss17 = explicit_slot 16
    ss18 = explicit_slot 24
    ss19 = explicit_slot 4
    ss20 = explicit_slot 4
    ss21 = explicit_slot 4
    ss22 = explicit_slot 4
    ss23 = explicit_slot 16
    ss24 = explicit_slot 16
    ss25 = explicit_slot 16
    ss26 = explicit_slot 16
    ss27 = explicit_slot 48
    ss28 = explicit_slot 16
    ss29 = explicit_slot 16
    ss30 = explicit_slot 32
    ss31 = explicit_slot 16
    ss32 = explicit_slot 8
    ss33 = explicit_slot 8
    ss34 = explicit_slot 16
    ss35 = explicit_slot 16
    ss36 = explicit_slot 16
    ss37 = explicit_slot 48
    ss38 = explicit_slot 16
    ss39 = explicit_slot 16
    ss40 = explicit_slot 32
    ss41 = explicit_slot 16
    ss42 = explicit_slot 8
    ss43 = explicit_slot 8
    ss44 = explicit_slot 16
    ss45 = explicit_slot 16
    ss46 = explicit_slot 16
    ss47 = explicit_slot 16
    ss48 = explicit_slot 16
    ss49 = explicit_slot 16
    ss50 = explicit_slot 16
    ss51 = explicit_slot 8
    ss52 = explicit_slot 4
    ss53 = explicit_slot 4
    ss54 = explicit_slot 16
    ss55 = explicit_slot 16
    ss56 = explicit_slot 16
    ss57 = explicit_slot 2
    ss58 = explicit_slot 4
    ss59 = explicit_slot 2
    ss60 = explicit_slot 16
    ss61 = explicit_slot 16
    ss62 = explicit_slot 16
    ss63 = explicit_slot 16
    ss64 = explicit_slot 16
    ss65 = explicit_slot 16
    ss66 = explicit_slot 16
    ss67 = explicit_slot 16
    ss68 = explicit_slot 8
    ss69 = explicit_slot 16
    ss70 = explicit_slot 16
    ss71 = explicit_slot 48
    ss72 = explicit_slot 16
    ss73 = explicit_slot 16
    ss74 = explicit_slot 32
    ss75 = explicit_slot 16
    ss76 = explicit_slot 8
    ss77 = explicit_slot 8
    ss78 = explicit_slot 16
    ss79 = explicit_slot 16
    ss80 = explicit_slot 16
    ss81 = explicit_slot 48
    ss82 = explicit_slot 16
    ss83 = explicit_slot 16
    ss84 = explicit_slot 32
    ss85 = explicit_slot 16
    ss86 = explicit_slot 8
    ss87 = explicit_slot 8
    ss88 = explicit_slot 16
    ss89 = explicit_slot 16
    ss90 = explicit_slot 4
    ss91 = explicit_slot 16
    ss92 = explicit_slot 16
    ss93 = explicit_slot 16
    ss94 = explicit_slot 16
    ss95 = explicit_slot 16
    ss96 = explicit_slot 16
    ss97 = explicit_slot 2
    ss98 = explicit_slot 16
    ss99 = explicit_slot 16
    ss100 = explicit_slot 16
    ss101 = explicit_slot 16
    ss102 = explicit_slot 16
    ss103 = explicit_slot 16
    ss104 = explicit_slot 8
    ss105 = explicit_slot 16
    ss106 = explicit_slot 16
    ss107 = explicit_slot 4
    ss108 = explicit_slot 16
    ss109 = explicit_slot 16
    ss110 = explicit_slot 16
    ss111 = explicit_slot 16
    ss112 = explicit_slot 4
    ss113 = explicit_slot 4
    ss114 = explicit_slot 4
    ss115 = explicit_slot 4
    ss116 = explicit_slot 16
    ss117 = explicit_slot 16
    ss118 = explicit_slot 16
    ss119 = explicit_slot 16
    ss120 = explicit_slot 16
    ss121 = explicit_slot 4
    ss122 = explicit_slot 4
    ss123 = explicit_slot 16
    ss124 = explicit_slot 16
    ss125 = explicit_slot 16
    ss126 = explicit_slot 2
    ss127 = explicit_slot 16
    ss128 = explicit_slot 16
    ss129 = explicit_slot 16
    ss130 = explicit_slot 16
    ss131 = explicit_slot 16
    ss132 = explicit_slot 4
    ss133 = explicit_slot 16
    ss134 = explicit_slot 16
    ss135 = explicit_slot 16
    ss136 = explicit_slot 16
    ss137 = explicit_slot 16
    ss138 = explicit_slot 16
    ss139 = explicit_slot 2
    ss140 = explicit_slot 16
    ss141 = explicit_slot 16
    ss142 = explicit_slot 16
    ss143 = explicit_slot 16
    ss144 = explicit_slot 4
    gv0 = symbol colocated u1:22
    gv1 = symbol colocated u1:23
    gv2 = symbol colocated u1:24
    gv3 = symbol colocated u1:23
    gv4 = symbol colocated u1:25
    gv5 = symbol colocated u1:23
    gv6 = symbol colocated u1:26
    gv7 = symbol colocated u1:23
    gv8 = symbol colocated u1:27
    gv9 = symbol colocated u1:23
    gv10 = symbol colocated u1:28
    gv11 = symbol colocated u1:23
    gv12 = symbol colocated u1:29
    gv13 = symbol colocated u1:30
    gv14 = symbol colocated u1:31
    gv15 = symbol colocated u1:23
    gv16 = symbol colocated u1:29
    gv17 = symbol colocated u1:32
    gv18 = symbol colocated u1:32
    gv19 = symbol colocated u1:32
    gv20 = symbol colocated u1:32
    gv21 = symbol colocated u1:32
    gv22 = symbol colocated u1:33
    gv23 = symbol colocated u1:34
    gv24 = symbol colocated u1:23
    gv25 = symbol colocated u1:35
    gv26 = symbol colocated u1:36
    gv27 = symbol colocated u1:23
    gv28 = symbol colocated u1:29
    gv29 = symbol colocated u1:32
    gv30 = symbol colocated u1:37
    gv31 = symbol colocated u1:38
    gv32 = symbol colocated u1:30
    gv33 = symbol colocated u1:32
    gv34 = symbol colocated u1:32
    gv35 = symbol colocated u1:29
    gv36 = symbol colocated u1:32
    gv37 = symbol colocated u1:30
    gv38 = symbol colocated u1:32
    gv39 = symbol colocated u1:39
    gv40 = symbol colocated u1:40
    gv41 = symbol colocated u1:41
    gv42 = symbol colocated u1:23
    gv43 = symbol colocated u1:29
    gv44 = symbol colocated u1:42
    gv45 = symbol colocated u1:29
    gv46 = symbol colocated u1:30
    gv47 = symbol colocated u1:29
    gv48 = symbol colocated u1:30
    gv49 = symbol colocated u1:32
    gv50 = symbol colocated u1:43
    gv51 = symbol colocated u1:44
    gv52 = symbol colocated u1:45
    gv53 = symbol colocated u1:23
    gv54 = symbol colocated u1:46
    gv55 = symbol colocated u1:47
    gv56 = symbol colocated u1:48
    gv57 = symbol colocated u1:23
    gv58 = symbol colocated u1:32
    gv59 = symbol colocated u1:39
    gv60 = symbol colocated u1:49
    gv61 = symbol colocated u1:49
    gv62 = symbol colocated u1:49
    gv63 = symbol colocated u1:38
    gv64 = symbol colocated u1:30
    gv65 = symbol colocated u1:32
    gv66 = symbol colocated u1:50
    gv67 = symbol colocated u1:23
    gv68 = symbol colocated u1:29
    gv69 = symbol colocated u1:51
    gv70 = symbol colocated u1:29
    gv71 = symbol colocated u1:30
    gv72 = symbol colocated u1:32
    gv73 = symbol colocated u1:49
    gv74 = symbol colocated u1:32
    sig0 = (i64) system_v
    sig1 = (i64) system_v
    sig2 = (i64) system_v
    sig3 = (i64) system_v
    sig4 = (i64) system_v
    sig5 = (i64) system_v
    sig6 = (i64, i64, i64) system_v
    sig7 = (i64) -> i8 system_v
    sig8 = (i64) system_v
    sig9 = (i64) system_v
    sig10 = (i64, i64, i64) system_v
    sig11 = (i64) -> i8 system_v
    sig12 = (i64) system_v
    sig13 = (i64) system_v
    sig14 = (i64) -> i64 system_v
    sig15 = (i64) system_v
    sig16 = (i64) system_v
    sig17 = (i64) system_v
    sig18 = (i64) system_v
    sig19 = (i64) system_v
    sig20 = (i64) system_v
    sig21 = (i64) system_v
    sig22 = (i64, i64) system_v
    sig23 = (i64) system_v
    sig24 = (i64, i64, i16) system_v
    sig25 = (i64, i64, i16) system_v
    sig26 = (i64) system_v
    sig27 = (i64) system_v
    sig28 = (i64) system_v
    sig29 = (i64) system_v
    sig30 = (i64, i16, i16) system_v
    sig31 = (i64, i64, i64) system_v
    sig32 = (i64, i64, i64) system_v
    sig33 = (i64, i64, i64) system_v
    sig34 = (i64, i64) -> i8 system_v
    sig35 = (i64, i64, i64) system_v
    sig36 = (i64, i64) -> i8 system_v
    sig37 = (i64, i64, i64) system_v
    sig38 = (i64, i64, i64) system_v
    sig39 = (i64, i64) system_v
    sig40 = (i64) system_v
    sig41 = (i64, i64) -> i8 system_v
    sig42 = (i64, i64, i64) system_v
    sig43 = (i64, i64) -> i8 system_v
    sig44 = (i64, i64, i64) system_v
    sig45 = (i64, i64, i64) system_v
    sig46 = (i64, i64) system_v
    sig47 = (i64) system_v
    sig48 = (i64) system_v
    sig49 = (i64) system_v
    sig50 = (i64) system_v
    sig51 = (i64) system_v
    sig52 = (i64) system_v
    sig53 = (i64) system_v
    sig54 = (i64, i32) system_v
    sig55 = (i64) system_v
    sig56 = (i64) system_v
    sig57 = (i64) system_v
    sig58 = (i64) system_v
    sig59 = (i64) system_v
    sig60 = (i64) system_v
    sig61 = (i64) system_v
    sig62 = (i64) system_v
    sig63 = (i64) system_v
    sig64 = (i64) system_v
    sig65 = (i64) system_v
    sig66 = (i64) system_v
    sig67 = (i64) system_v
    sig68 = (i64) system_v
    sig69 = (i64) system_v
    sig70 = (i64, i64, i64) system_v
    sig71 = (i64) system_v
    sig72 = (i64, i64, i16, i64, i64, i64, i64, i64) system_v
    sig73 = (i64, i64) -> i8 system_v
    sig74 = (i64, i64, i64) system_v
    sig75 = (i64, i64) -> i8 system_v
    sig76 = (i64, i64, i64) system_v
    sig77 = (i64, i64, i64) system_v
    sig78 = (i64, i64) system_v
    sig79 = (i64) system_v
    sig80 = (i64, i64) -> i8 system_v
    sig81 = (i64, i64, i64) system_v
    sig82 = (i64, i64) -> i8 system_v
    sig83 = (i64, i64, i64) system_v
    sig84 = (i64, i64, i64) system_v
    sig85 = (i64, i64) system_v
    sig86 = (i64) system_v
    sig87 = (i64) system_v
    sig88 = (i64) system_v
    sig89 = (i64) system_v
    sig90 = (i64) system_v
    sig91 = (i64) system_v
    sig92 = (i64) system_v
    sig93 = (i64) system_v
    sig94 = (i64) system_v
    sig95 = (i64) system_v
    sig96 = (i64) system_v
    sig97 = (i64) system_v
    sig98 = (i64) system_v
    sig99 = (i64) system_v
    sig100 = (i64) system_v
    sig101 = (i64, i64, i64) system_v
    sig102 = (i64) system_v
    sig103 = (i64) system_v
    sig104 = (i64, i64, i16, i64, i64, i64, i64, i64) system_v
    sig105 = (i64) system_v
    fn0 = u0:83 sig0
    fn1 = u0:13 sig1
    fn2 = u0:83 sig2
    fn3 = u0:13 sig3
    fn4 = u0:83 sig4
    fn5 = u0:13 sig5
    fn6 = u0:84 sig6
    fn7 = u0:85 sig7
    fn8 = u0:83 sig8
    fn9 = u0:13 sig9
    fn10 = u0:86 sig10
    fn11 = u0:85 sig11
    fn12 = u0:83 sig12
    fn13 = u0:13 sig13
    fn14 = u0:16 sig14
    fn15 = u0:83 sig15
    fn16 = u0:13 sig16
    fn17 = u0:13 sig17
    fn18 = u0:13 sig18
    fn19 = u0:83 sig19
    fn20 = u0:13 sig20
    fn21 = u0:13 sig21
    fn22 = u0:87 sig22
    fn23 = u0:13 sig23
    fn24 = u0:88 sig24
    fn25 = u0:88 sig25
    fn26 = u0:13 sig26
    fn27 = u0:13 sig27
    fn28 = u0:13 sig28
    fn29 = u0:13 sig29
    fn30 = u0:89 sig30
    fn31 = u0:90 sig31
    fn32 = u0:90 sig32
    fn33 = u0:90 sig33
    fn34 = u0:91 sig34
    fn35 = u0:92 sig35
    fn36 = u0:91 sig36
    fn37 = u0:92 sig37
    fn38 = u0:11 sig38
    fn39 = u0:12 sig39
    fn40 = u0:13 sig40
    fn41 = u0:91 sig41
    fn42 = u0:92 sig42
    fn43 = u0:91 sig43
    fn44 = u0:92 sig44
    fn45 = u0:11 sig45
    fn46 = u0:12 sig46
    fn47 = u0:13 sig47
    fn48 = u0:13 sig48
    fn49 = u0:13 sig49
    fn50 = u0:13 sig50
    fn51 = u0:13 sig51
    fn52 = u0:13 sig52
    fn53 = u0:13 sig53
    fn54 = u0:93 sig54
    fn55 = u0:13 sig55
    fn56 = u0:13 sig56
    fn57 = u0:13 sig57
    fn58 = u0:13 sig58
    fn59 = u0:13 sig59
    fn60 = u0:13 sig60
    fn61 = u0:13 sig61
    fn62 = u0:83 sig62
    fn63 = u0:13 sig63
    fn64 = u0:13 sig64
    fn65 = u0:13 sig65
    fn66 = u0:13 sig66
    fn67 = u0:13 sig67
    fn68 = u0:13 sig68
    fn69 = u0:13 sig69
    fn70 = u0:94 sig70
    fn71 = u0:13 sig71
    fn72 = u0:95 sig72
    fn73 = u0:96 sig73
    fn74 = u0:97 sig74
    fn75 = u0:96 sig75
    fn76 = u0:97 sig76
    fn77 = u0:11 sig77
    fn78 = u0:12 sig78
    fn79 = u0:13 sig79
    fn80 = u0:91 sig80
    fn81 = u0:92 sig81
    fn82 = u0:91 sig82
    fn83 = u0:92 sig83
    fn84 = u0:11 sig84
    fn85 = u0:12 sig85
    fn86 = u0:13 sig86
    fn87 = u0:13 sig87
    fn88 = u0:13 sig88
    fn89 = u0:13 sig89
    fn90 = u0:13 sig90
    fn91 = u0:13 sig91
    fn92 = u0:13 sig92
    fn93 = u0:13 sig93
    fn94 = u0:13 sig94
    fn95 = u0:83 sig95
    fn96 = u0:13 sig96
    fn97 = u0:13 sig97
    fn98 = u0:13 sig98
    fn99 = u0:13 sig99
    fn100 = u0:13 sig100
    fn101 = u0:94 sig101
    fn102 = u0:13 sig102
    fn103 = u0:13 sig103
    fn104 = u0:95 sig104

ebb0(v0: i64, v1: i64, v2: i64):
    v113 -> v1
    v124 -> v1
    v136 -> v1
    v148 -> v1
    v160 -> v1
    v185 -> v1
    v222 -> v1
    v237 -> v1
    v241 -> v1
    v256 -> v1
    v262 -> v1
    v105 = iconst.i64 0
    v107 = iconst.i64 0
    v108 = iconst.i64 0
    jump ebb1

ebb1:
    v121 = iconst.i32 0
    brz v121, ebb3
    jump ebb2

ebb2:
    trap user65535

ebb3:
    v133 = iconst.i32 0
    brz v133, ebb5
    jump ebb4

ebb4:
    trap user65535

ebb5:
    v145 = iconst.i32 0
    brz v145, ebb7
    jump ebb6

ebb6:
    trap user65535

ebb7:
    jump ebb8

ebb8:
    jump ebb9

ebb9:
    v157 = iconst.i32 0
    brz v157, ebb11
    jump ebb10

ebb10:
    trap user65535

ebb11:
    jump ebb12

ebb12:
    jump ebb13

ebb13:
    v169 = iconst.i32 0
    brz v169, ebb15
    jump ebb14

ebb14:
    trap user65535

ebb15:
    jump ebb16

ebb16:
    v182 = iconst.i32 0
    brz v182, ebb18
    jump ebb17

ebb17:
    trap user65535

ebb18:
    v197 = iconst.i32 0
    brz v197, ebb19
    trap user0

ebb19:
    v210 = iconst.i32 0
    brz v210, ebb20
    trap user0

ebb20:
    v219 = iconst.i32 0
    brz v219, ebb22
    jump ebb21

ebb21:
    trap user65535

ebb22:
    v234 = iconst.i32 0
    brz v234, ebb23
    trap user0

ebb23:
    jump ebb24

ebb24:
    v253 = iconst.i32 0
    brz v253, ebb25
    trap user0

ebb25:
    jump ebb26

ebb26:
    jump ebb27

ebb27:
    v279 = iconst.i32 0
    brz v279, ebb28
    trap user0

ebb28:
    v290 = iconst.i32 0
    brz v290, ebb29
    trap user0

ebb29:
    v303 = iconst.i32 0
    brz v303, ebb30
    trap user0

ebb30:
    v314 = iconst.i32 0
    brz v314, ebb31
    trap user0

ebb31:
    jump ebb32

ebb32:
    v318 = iconst.i16 0
    v1007 -> v318
    jump ebb33

ebb33:
    jump ebb34

ebb34:
    jump ebb35

ebb35:
    jump ebb36

ebb36:
    v337 = iconst.i32 0
    brz v337, ebb38
    jump ebb37

ebb37:
    jump ebb39

ebb38:
    jump ebb42

ebb39:
    jump ebb40

ebb40:
    jump ebb41

ebb41:
    trap user65535

ebb42:
    jump ebb43

ebb43:
    v377 = iconst.i32 0
    brz v377, ebb45
    jump ebb44

ebb44:
    jump ebb46

ebb45:
    jump ebb49(v1007)

ebb46:
    jump ebb47

ebb47:
    jump ebb48

ebb48:
    trap user65535

ebb49(v1006: i16):
    v486 -> v1006
    v412 = iconst.i32 0
    brz v412, ebb50
    trap user0

ebb50:
    v414 = iconst.i64 0
    v439 -> v414
    v452 -> v414
    v478 -> v414
    v508 -> v414
    v425 = iconst.i32 0
    brz v425, ebb51
    trap user0

ebb51:
    v433 = iconst.i32 0
    brz v433, ebb52
    trap user0

ebb52:
    v438 = iconst.i64 0
    v453 -> v438
    v521 -> v438
    v448 = iconst.i32 0
    brz v448, ebb53
    trap user0

ebb53:
    v451 = iconst.i32 0
    v480 -> v451
    v551 -> v451
    v463 = iconst.i32 0
    brz v463, ebb54
    trap user0

ebb54:
    v475 = iconst.i32 0
    brz v475, ebb55
    trap user0

ebb55:
    v479 = iconst.i64 0
    jump ebb56

ebb56:
    v481 = iconst.i8 0
    v548 -> v481
    v484 = iconst.i64 0
    v494 = iconst.i32 0
    brz v494, ebb57
    trap user0

ebb57:
    v505 = iconst.i32 0
    brz v505, ebb58
    trap user0

ebb58:
    v507 = iconst.i16 0
    v518 = iconst.i32 0
    brz v518, ebb59
    trap user0

ebb59:
    v520 = iconst.i64 0
    v546 -> v520
    v531 = iconst.i32 0
    brz v531, ebb60
    trap user0

ebb60:
    v543 = iconst.i32 0
    brz v543, ebb61
    trap user0

ebb61:
    v547 = iconst.i64 0
    jump ebb62(v551, v484, v521, v479, v520, v507, v508, v548, v547)

ebb62(v552: i32, v1009: i64, v1013: i64, v1016: i64, v1019: i64, v1022: i16, v1025: i64, v1028: i8, v1033: i64):
    v559 -> v552
    v562 -> v552
    v569 -> v552
    v596 -> v1009
    v605 -> v1009
    v609 -> v1009
    v1008 -> v1009
    v624 -> v1013
    v654 -> v1013
    v1012 -> v1013
    v1014 -> v1013
    v1041 -> v1013
    v636 -> v1016
    v1015 -> v1016
    v1017 -> v1016
    v1030 -> v1016
    v648 -> v1019
    v676 -> v1019
    v693 -> v1019
    v1018 -> v1019
    v1020 -> v1019
    v674 -> v1022
    v691 -> v1022
    v1021 -> v1022
    v1023 -> v1022
    v1054 -> v1022
    v677 -> v1025
    v1024 -> v1025
    v1026 -> v1025
    v1059 -> v1025
    v696 -> v1028
    v1027 -> v1028
    v1029 -> v1028
    v1031 -> v1033
    v1032 -> v1033
    v1034 -> v1033
    v557 = iconst.i32 0
    brz v557, ebb63
    trap user0

ebb63:
    v561 = iconst.i32 0
    v574 -> v561
    v567 = iconst.i32 0
    brz v567, ebb64
    trap user0

ebb64:
    jump ebb65

ebb65:
    v581 = iconst.i32 0
    brz v581, ebb67
    jump ebb66

ebb66:
    trap user65535

ebb67:
    jump ebb68(v574)

ebb68(v584: i32):
    v594 = iconst.i32 0
    brz v594, ebb69
    trap user0

ebb69:
    v601 = iconst.i32 0
    brnz v601, ebb70
    trap user0

ebb70:
    v619 = iconst.i32 0
    brz v619, ebb71
    trap user0

ebb71:
    v621 = iconst.i64 0
    v668 -> v621
    v695 -> v621
    v1010 -> v621
    v1046 -> v621
    v633 = iconst.i32 0
    brz v633, ebb72
    trap user0

ebb72:
    v645 = iconst.i32 0
    brz v645, ebb73
    trap user0

ebb73:
    v651 = iconst.i32 0
    brz v651, ebb80
    jump ebb74

ebb74:
    trap user0

ebb79:
    return

ebb80:
    jump ebb81

ebb81:
    jump ebb82

ebb82:
    v715 = iconst.i32 0
    brz v715, ebb84
    jump ebb83

ebb83:
    jump ebb85

ebb84:
    jump ebb88

ebb85:
    jump ebb86

ebb86:
    jump ebb87

ebb87:
    trap user65535

ebb88:
    jump ebb89

ebb89:
    v754 = iconst.i32 0
    brz v754, ebb91
    jump ebb90

ebb90:
    jump ebb92

ebb91:
    jump ebb95(v1030, v1031, v1041, v1046, v1054, v1059)

ebb92:
    jump ebb93

ebb93:
    jump ebb94

ebb94:
    trap user65535

ebb95(v779: i64, v780: i64, v1040: i64, v1045: i64, v1053: i16, v1058: i64):
    v781 = iconst.i64 0
    jump ebb99(v779, v780, v781, v1040, v1045, v1053, v1058)

ebb99(v804: i64, v1035: i64, v1037: i64, v1039: i64, v1044: i64, v1052: i16, v1057: i64):
    v817 -> v1035
    v830 -> v1037
    v844 -> v1039
    v857 -> v1039
    v939 -> v1039
    v1042 -> v1039
    v1050 -> v1039
    v908 -> v1044
    v917 -> v1044
    v921 -> v1044
    v1043 -> v1044
    v960 -> v1052
    v990 -> v1052
    v1051 -> v1052
    v1055 -> v1052
    v963 -> v1057
    v1056 -> v1057
    v1060 -> v1057
    v814 = iconst.i32 0
    brz v814, ebb100
    trap user0

ebb100:
    v827 = iconst.i32 0
    brz v827, ebb101
    trap user0

ebb101:
    v829 = iconst.i64 0
    v935 -> v829
    v962 -> v829
    v992 -> v829
    v1036 -> v829
    v1049 -> v829
    v840 = iconst.i32 0
    brz v840, ebb102
    trap user0

ebb102:
    v842 = iconst.i64 0
    v976 -> v842
    v989 -> v842
    v1038 -> v842
    v1061 -> v842
    v853 = iconst.i32 0
    brz v853, ebb103
    trap user0

ebb103:
    v855 = iconst.i64 0
    v886 -> v855
    v867 = iconst.i32 0
    brz v867, ebb104
    trap user0

ebb104:
    v879 = iconst.i32 0
    brz v879, ebb105
    trap user0

ebb105:
    v883 = iconst.i64 0
    v934 -> v883
    v961 -> v883
    v991 -> v883
    v1005 -> v883
    v1048 -> v883
    jump ebb106

ebb106:
    v893 = iconst.i32 0
    brz v893, ebb108
    jump ebb107

ebb107:
    trap user65535

ebb108:
    jump ebb109(v886)

ebb109(v896: i64):
    v906 = iconst.i32 0
    brz v906, ebb110
    trap user0

ebb110:
    v913 = iconst.i32 0
    brnz v913, ebb111
    trap user0

ebb111:
    v931 = iconst.i32 0
    brz v931, ebb112
    trap user0

ebb112:
    v933 = iconst.i64 0
    v954 -> v933
    v1047 -> v933
    v938 = iconst.i32 0
    brz v938, ebb119
    jump ebb113

ebb113:
    v949 = iconst.i32 0
    brz v949, ebb114
    trap user0

ebb114:
    v951 = iconst.i64 0
    v988 -> v951
    v955 = iconst.i64 0
    call fn101(v107, v108, v955)
    jump ebb115

ebb115:
    v973 = iconst.i32 0
    brz v973, ebb116
    trap user0

ebb116:
    v985 = iconst.i32 0
    brz v985, ebb117
    trap user0

ebb117:
    v987 = iconst.i64 0
    call fn104(v0, v105, v990, v991, v992, v987, v988, v989)
    jump ebb118

ebb118:
    jump ebb79

ebb119:
    v1002 = iconst.i32 0
    brz v1002, ebb120
    trap user0

ebb120:
    jump ebb99(v1005, v1036, v1038, v1042, v1047, v1055, v1060)
}
```

</details>

**Comment by julian-seward1** [comment]
@bjorn3, do you have here, in the stack slots, values of type other than i32, i64, f32 or f64?  #773 only deals with those four types and I suspect it may have failed here because something else turned up.

**Comment by bjorn3** [comment]
There are i8 and i16 values which may have been spilled.

**Comment by julian-seward1** [comment]
Sorry about this.  I'll investigate in the morning.

**Comment by bjorn3** [comment]
Thanks

**Comment by julian-seward1** [comment]
@bjorn3, can you try this?  It works OK for me on your test case.
```
diff --git a/cranelift-codegen/meta-python/isa/x86/encodings.py b/cranelift-codegen/meta-python/isa/x86/encodings.py
index bfd1f6ef..2fdc9346 100644
--- a/cranelift-codegen/meta-python/isa/x86/encodings.py
+++ b/cranelift-codegen/meta-python/isa/x86/encodings.py
@@ -341,9 +341,16 @@ enc_x86_64(x86.pop.i64, r.popq, 0x58)
 X86_64.enc(base.copy_special, *r.copysp.rex(0x89, w=1))
 X86_32.enc(base.copy_special, *r.copysp(0x89))
 
-# Stack-slot-to-the-same-stack-slot copy, which is guaranteed to turn into a no-op.
+# Stack-slot-to-the-same-stack-slot copy, which is guaranteed to turn
+# into a no-op.  Ideally we could to make this encoding available for
+# all types, and write `base.copy_nop.any`, but it appears that the
+# controlling type variable must not polymorphic.  So we make do with
+# the following limited set, and guard the generating transformation in
+# regalloc/reload.rs accordingly.
 X86_64.enc(base.copy_nop.i64, r.stacknull, 0)
 X86_64.enc(base.copy_nop.i32, r.stacknull, 0)
+X86_64.enc(base.copy_nop.i16, r.stacknull, 0)
+X86_64.enc(base.copy_nop.i8,  r.stacknull, 0)
 X86_64.enc(base.copy_nop.f64, r.stacknull, 0)
 X86_64.enc(base.copy_nop.f32, r.stacknull, 0)
 
diff --git a/cranelift-codegen/src/regalloc/reload.rs b/cranelift-codegen/src/regalloc/reload.rs
index cc6aae59..c94e8e48 100644
--- a/cranelift-codegen/src/regalloc/reload.rs
+++ b/cranelift-codegen/src/regalloc/reload.rs
@@ -222,15 +222,28 @@ impl<'a> Context<'a> {
         {
             let dst_vals = self.cur.func.dfg.inst_results(inst);
             if dst_vals.len() == 1 {
+                let dst_val = dst_vals[0];
                 let can_transform = match (
                     self.cur.func.locations[arg],
-                    self.cur.func.locations[dst_vals[0]],
+                    self.cur.func.locations[dst_val],
                 ) {
-                    (ValueLoc::Stack(src_slot), ValueLoc::Stack(dst_slot)) => src_slot == dst_slot,
+                    (ValueLoc::Stack(src_slot), ValueLoc::Stack(dst_slot)) => {
+                        src_slot == dst_slot && {
+                            let src_ty = self.cur.func.dfg.value_type(arg);
+                            let dst_ty = self.cur.func.dfg.value_type(dst_val);
+                            debug_assert!(src_ty == dst_ty);
+                            // This limits the transformation to copies of the
+                            // types: I64 I32 I16 I8 F64 and F32, since that's
+                            // the set of `copy_nop` encodings available.
+                            src_ty.is_int() || src_ty.is_float()
+                        }
+                    }
                     _ => false,
                 };
                 if can_transform {
-                    // Convert the instruction into a `copy_nop`.
+                    // Convert the instruction into a `copy_nop`.  The
+                    // assertion will fail if there is no encoding for
+                    // `copy_nop` at the required type.
                     self.cur.func.dfg.replace(inst).copy_nop(arg);
                     let ok = self.cur.func.update_encoding(inst, self.cur.isa).is_ok();
                     debug_assert!(ok);
```

**Comment by bjorn3** [comment]
Fixed in #784.
